### PR TITLE
Only lookup names for relevant ERC20 transfers

### PR DESCRIPTION
**Pull request by 0xDaedalus**
This PR resolves the issue of name service lookups temporarily bringing the application to a halt.

### To Test
- Load the extension, add an account, open the background logs
- Go do something else for ~4 minutes
- Come back - focus all your energy on testing this ticket
- [ ] The background logs should not be full of `ERR_INSUFFICIENT_RESOURCES` errors as pictured below:

image

### CODE REVIEW

**Review by greg-nagy — APPROVED** [review]
let's 🚢

File: background/services/enrichment/utils.ts [hunk 1]
```diff
@@ -81,3 +82,18 @@ export function getDistinctRecipentAddressesFromERC20Logs(
...
   return [...new Set([...logs.map(({ recipientAddress }) => recipientAddress)])]
 }
+
+export const getERC20LogsForAddresses = (
+  logs: ERC20TransferLog[],
+  addresses: string[]
+): ERC20TransferLog[] => {
+  const relevantAddresses = Object.fromEntries(
+    addresses.map((address) => [address, true])
+  )
```
Comment (Shadowfiend) [inline background/services/enrichment/utils.ts:92]: You can just use a `Set` here fwiw.

**Review by Shadowfiend — COMMENTED** [review]
Solid find 😅

**Comment by SJQuality** [comment]
Lots of errors, but not those errors:
image

**Pull request merged**